src/supporting-data/citations/components/LiteratureCitation.tsx

```diff
@@ -184,14 +190,25 @@ export const JournalInfo: FC<JournalInfoProps> = ({
     submissionDatabase,
     citationType,
     locator,
+    editors,
+    bookName,
+    publisher,
+    address,
+    patentNumber,
   },
 }) => {
-  const name = journal || doiId;
+  let name = journal || doiId;
+
+  if (citationType === 'book' && bookName) {
+    name =
+      editors !== undefined ? `${editors?.join(',')}; ${bookName}` : bookName;
+  }
+
   let page = null;
   if (firstPage) {
     page = firstPage;
     if (lastPage) {
-      page += `-${lastPage}`;
+      page += `-${lastPage} `;
```

Comment:
I would not put the extra space there. What if you only have a first page and no last page?